src/core/service.c

```diff
@@ -246,6 +256,20 @@ static void service_reset_watchdog(Service *s) {
         service_start_watchdog(s);
 }
 
+static void service_reset_watchdog_timeout(Service *s, usec_t watchdog_override_usec) {
+        assert(s);
+
+        if (watchdog_override_usec == 0)
+                return ;
```

Comment:
I am pretty sure there should be a way to fully disable the watchdog, but this makes that impossible no?